### PR TITLE
Parallel start

### PR DESCRIPTION
**Pull request by starkmsu**

### CODE REVIEW

**Review by OttoVT — DISMISSED** [review]

**Review by tarurar — DISMISSED** [review]
Better to have it reviewed by someone else who'd contributed before to assess the sense of changes.

File: src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs [hunk 1]
```diff
@@ -14,11 +15,10 @@ public class RabbitMqTransportFactory : ITransportFactory
         private readonly ILogFactory _logFactory;
         private readonly bool m_ShuffleBrokers;
         private readonly TimeSpan? m_AutomaticRecoveryInterval;
+        private readonly Dictionary<TransportInfo, RabbitMqTransport> _transports = new Dictionary<TransportInfo, RabbitMqTransport>();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs]: Why not ConcurrentDictionary + Lazy? https://andrewlock.net/making-getoradd-on-concurrentdictionary-thread-safe-using-lazy/

File: src/Lykke.Messaging/ProcessingGroupManager.cs [hunk 2]
```diff
@@ -178,27 +174,19 @@ private string GetSessionName(ProcessingGroup processingGroup, int priority)
...
         {
-            ProcessingGroup group;
-            lock (m_ProcessingGroups)
-            {
-                if (m_ProcessingGroups.TryGetValue(processingGroup, out group)) 
-                    return group;
+            if (m_ProcessingGroups.TryGetValue(processingGroup, out var group))
+                return group;
+
+            var info = m_ProcessingGroupInfos.GetOrAdd(processingGroup, new ProcessingGroupInfo());
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Messaging/ProcessingGroupManager.cs:180]: Is ProcessingGroupInfo instantiation heavy operation?

File: src/Lykke.Messaging/ProcessingGroupManager.cs [hunk 3]
```diff
@@ -178,27 +174,19 @@ private string GetSessionName(ProcessingGroup processingGroup, int priority)
...
+            group = m_ProcessingGroups.GetOrAdd(processingGroup, new ProcessingGroup(processingGroup, info));
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Messaging/ProcessingGroupManager.cs:181]: Is ProcessingGroup instantiation heavy operation?

Comment (starkmsu) [reply src/Lykke.Messaging/ProcessingGroupManager.cs:181]: No

File: src/Lykke.Messaging/ProcessingGroupManager.cs (hunk 2 above)
Comment (starkmsu) [reply src/Lykke.Messaging/ProcessingGroupManager.cs:180]: No

File: src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs [hunk 4]
```diff
@@ -84,31 +84,22 @@ public ITransport Create(ILog log, TransportInfo transportInfo, Action onFailure
...
-                    return result;
-
-                var brokers = transportInfo.Broker
+                var brokers = ti.Broker
                     .Split(',')
                     .Select(b => b.Trim())
                     .ToArray();
 
-                result = new RabbitMqTransport(
+                return new RabbitMqTransport(
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs:93]: Is ```new RabbitMqTransport()``` heavy operation? If so, use Lazy together with ConcurrentDictionary, as I've suggested above.

Comment (starkmsu) [reply src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs:93]: No, it's not heavy

**Review by KonstantinRyazantsev — APPROVED** [review]

File: src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs (hunk 4 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs:93]: Ok, but note that multiple ```RabbitMqTransport``` instances can be created for the same ```transportInfo```

Comment (starkmsu) [reply src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs:93]: That's ok

**Pull request merged**